Add/update docs for new trust-manager features (#1351)
cert-manager/website: content/docs/trust/trust-manager/README.md

- `inLine` - a manually specified string containing at least one certificate
- `useDefaultCAs` - usually, a bundle of publicly trusted certificates

`ConfigMap` is the default target type, but as of v0.7.0 trust-manager also supports `Secret` resources as targets.

Support for `Secret` targets must be explicitly enabled in the trust-manager controller; see details below under "Enable Secret targets".

> All sources and target options are documented in the trust-manager [API reference documentation](./api-reference.md).

Author: @SgtCoDFish Do we have CI to update API docs in this repo? It seems like it currently misses the v0.7.0 extensions of the Bundle API.

Reviewer: It requires a [manual bump here](https://github.com/cert-manager/website/blob/master/scripts/gendocs/generate-trust-manager#L62).

It probably _shouldn't_ require a manual bump, but it does #:grin:

#### Targets

All `Bundle` targets are written to `ConfigMap`s (and/or `Secret`s) whose name matches that of the
`Bundle`, and every target has a PEM-formatted bundle included.

Users can also optionally choose to write JKS/PKCS#12 formatted binary trust store(s) to targets.

> JKS has been supported since v0.5.0, and PKCS#12 since v0.7.0.

Author: > [spelling]     content/docs/trust/trust-manager/README.md
[spelling]       106 | nce v0.5.0, and PKCS#12 since v0.7.0.

Why? How to fix this? Add "v0.7.0." to the dictionary?

Reviewer: That'll probably be it yeah 😦 `./scripts/verify` should show locally with highlighting what specifically is broken

We understand that any binary trust store require a password, even though trust bundles don't contain
secrets. Both JKS and PKCS#12 uses weak encryption primitives, so a trust store (or keystore) will NOT

> be protected by a password alone, and needs to be protected by additional measures.

Author: I am not sure if I agree with this suggestion. Users want to use passwords on trust stores to protect the **integrity** of the trust store. We all can agree that a trust store doesn't contain anything secret, but some think it's important to use a secret password - even in Kubernetes. I am open to rewording here, it's just that I find the suggestion a false statement. 😉

> be protected by a password alone, and needs to be protected by additional measures.

Author: Thanks Ash! Your suggestion was so good that I included you as a co-author to this PR. ❤️ I hope Tim also think it's ok now.

- `inLine` - a manually specified string containing at least one certificate
- `useDefaultCAs` - usually, a bundle of publicly trusted certificates

> `ConfigMap` is the default target resource type, but trust-manager also supports `Secret` as targets
> (since trust-manager v0.7.0), but that feature has to be explicitly enabled on the controller.

Reviewer: suggestion:

```suggestion
`ConfigMap` is the default target type, but as of v0.7.0 trust-manager also supports `Secret` resources as targets.

Support for `Secret` targets must be explicitly enabled in the trust-manager controller; see details below under "Enable Secret targets".
```

All sources and target options are documented in the trust-manager [API reference documentation](./api-reference.md).

#### Targets

All `Bundle` targets are written to `ConfigMap`s (and/or `Secret`s) whose name matches that of the
`Bundle`, and every target has a PEM-formatted bundle included.

> Users can also optionally choose to write JKS/PKCS#12 formatted binary trust store(s) to the target.
> JKS is supported since trust-manager v0.5.0, while v0.7.0 add support for PKCS#12.

Reviewer: suggestion:

```suggestion
Users can also optionally choose to write JKS/PKCS#12 formatted binary trust store(s) to targets.
JKS has been supported since v0.5.0, and PKCS#12 since v0.7.0.
```

> We understand that any binary trust store require a password, even though trust bundles don't contain

Reviewer 2: TODO(@inteon): reread this section and add comments.

secrets. Both JKS and PKCS#12 uses weak encryption primitives, so a trust store (or keystore) will NOT
be protected by a password alone, and needs to be protected by additional measures.
Please understand that trust-manager provides these binary trust store formats to support legacy

> applications which do not support other formats.

Reviewer: suggestion:

> Please understand that trust-manager provides these binary trust store formats to support legacy
> applications which do not support other formats.

I don't think this is exclusively for legacy stuff. An app using JKS trust stores today probably doesn't have anything wrong with it - JKS is overcomplicated for the task for sure, but it'd work just fine and I'm sure it'll be supported basically forever. I'd remove these two lines personally.

Author: Agreed, removed.

> At present passwords to trust stores provided by trust-manager are hard-coded to `changeit` for JKS and
> `"""` (empty string; so called password-less) for PKCS#12 but this might be configurable on bundles
> in a future release of trust-manager.

Reviewer: suggestion:
 
```suggestion
At present, passwords for trust stores are hard-coded to `changeit` for JKS and `""` (the empty string or "passwordless") for PKCS#12. Future releases of trust-manager may make this configurable.
```
